src/main/java/com/celements/cleverreach/CleverReachRest.java

```diff
@@ -222,10 +286,48 @@ Response sendRequest(String path, Object data, String authHeader, SubmitMethod m
     return sendRequest(path, data, authHeader, method, connection.getBaseUrl());
   }
 
+  boolean isReadyToSendPut(Response response) {
+    if ((response != null) && response.hasEntity()) {
+      String content = response.readEntity(String.class);
+      try {
+        Optional<ResponseBodyObj> resultObj = Arrays.asList(new ObjectMapper()
+            .readValue(content, ResponseBodyObj.class)).stream().findFirst();
+        return resultObj.isPresent() && "1".equals(resultObj.get().value);
```

Comment:
improved chaining:
```suggestion
        return Optional.ofNullable(new ObjectMapper()
            .readValue(content, ResponseBodyObj.class))
            .map(responseObj -> "1".equals(responseObj.value))
            .orElse(false);
```
please verify in eclipse